src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2025, Hewlett Packard Enterprise Development LP */
+
+/*
+ * util_sds_check.c -- unit test for shutdown_state_check function
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "unittest.h"
+#include "ut_pmem2.h"
+#include "shutdown_state.h"
+#include "set.h"
+
+int os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush);
+/*
+ * os_part_deep_common -- XXX temporally workaround until we will have pmem2
+ *		integrated with common
+ */
+int
+os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush)
+{
+	return 0;
+}
+
+/* Test with curr_sds having non-zero values and pool_sds being zeroed out. */
+static int
+test_case_1()
+{
+	struct shutdown_state curr_sds = {1, 1, 57, {0}};
+	struct shutdown_state pool_sds = {0, 0, 0, {0}};
+	struct pool_replica rep;
+
+	int result = shutdown_state_check(&curr_sds, &pool_sds, &rep);
+	return (result == 0) ? 0 : 1;
```

Comment:
The return value is NOT the result of the test. It is the number of consumed arguments from an array of arguments for some reason the compiler accepted you not to need as an arguments of this function. 😄 

```suggestion
	return 0;
```

Please again see the core_log tests to see how the test case functions are defined. I am ok with keeping these test cases as is.

Reply:
Done.

Reply:
Nope.

Reply:
☝️

Reply:
Done.
<!-- Reviewable comment -OM6GfnM0e63VXGQ3rSd:b-896fix -->
<!-- Sent from Reviewable.io -->

Reply:
Done.
<!-- Reviewable comment -OM6GfnM0e63VXGQ3rSd:b-896fix -->
<!-- Sent from Reviewable.io -->